include/vctr/Containers/VctrBase.h

```diff
@@ -64,10 +64,49 @@ class VctrBase : private Config,
     auto* data() const { return storage.data(); }
 
     /** Returns the number of elements. This overload is a non-static function, used in case the extent is dynamic. */
-    size_t size() const requires (extent == std::dynamic_extent) { return storage.size(); }
+    constexpr size_t size() const noexcept requires (extent == std::dynamic_extent) { return storage.size(); }
 
     /** Returns the number of elements. This overload is a static function, used in case the extent is static. */
-    static constexpr size_t size() requires (extent != std::dynamic_extent) { return extent; }
+    static constexpr size_t size() noexcept requires (extent != std::dynamic_extent) { return extent; }
+
+    /** Returns the maximum number of elements the container is able to hold.
+
+        This overload is a non-static function, used in case the extent is dynamic.
+      */
+    constexpr size_t max_size() const noexcept requires (extent == std::dynamic_extent) { return storage.max_size(); }
```

Comment:
I think `max_size` is only specified in `std::vector`. So this code path would fail to compile for a `Span` with dynamic extent. [cppreference](https://en.cppreference.com/w/cpp/container/vector/max_size) says:

> Returns the maximum number of elements the container is able to hold due to system or library implementation limitations, i.e. [std::distance](http://en.cppreference.com/w/cpp/iterator/distance)(begin(), end()) for the largest container. [...] This value typically reflects the theoretical limit on the size of the container, at most [std::numeric_limits](http://en.cppreference.com/w/cpp/types/numeric_limits)<difference_type>::max().

So this is more about a theoretical system memory limitation for `std::vector` and probably rarely used in real world code. I'd propose to move this to the `Vector` implementation and drop the overload for the dynamic extent case